src/Controller/ActualityController.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Controller;
+
+use App\Entity\Article;
+use App\Entity\Image;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use App\Repository\ArticleRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Mapping\Entity;
+use Symfony\Component\HttpFoundation\Request;
+use App\Form\ArticleFormType;
+
+class ActualityController extends AbstractController
+{
+    /**
+     * @Route("/actualites", name="actuality")
+     */
+    public function showArticles(ArticleRepository $articleRepository): Response
+    {
+        $articles = $articleRepository->findBy(
+            ["section" => 'Actualites'],
+            ['id' => 'DESC'],
+        );
+        return $this->render('actuality/index.html.twig', ['articles' => $articles]);
+    }
+
+    /**
+     * Returning the 3 last articles posted, for the carousel in "Design métallique"
+     * @return object[]
+     */
+    public function showCarouselArticles(ArticleRepository $articleRepository): array
+    {
+        $articles = $articleRepository->findby(
+            ['section' => 'Actualites'],
+            ['id' => 'DESC'],
+            3,
+            0,
+        );
+        return $articles;
+    }
+
+    /**
+     * Editing an article, but can not delete image for now, or it goes out of the db too.
+     * @Route("/actualites/{id}/edit", name="actuality_edit")
+     */
+    public function editArticles(Request $request, Article $article): Response
+    {
+        /*$article->getImage()->removeElement($image);
+        $image->getArticles()->removeElement($article);*/
+        $form = $this->createForm(ArticleFormType::class, $article);
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->getDoctrine()->getManager()->flush();
```

Comment:
The symfony best practice is to use dependency injection to access your services

Reply:
Ok, modified.